### PR TITLE
Update default connection heartbeat length

### PR DESCRIPTION
**Pull request by jackgene**
Calculate default timeout_heartbeat_interval with minimum of 1s

This is as specified in the accompanying code comment "use a quarter of the actual configured timeout *but at minimum 1 second*".

### CODE REVIEW

**Comment by SeanTAllen** [comment]
Please remove the unrelated gitignore change.

**Comment by ponylang-main** [comment]
Hi @jackgene,

The changelog - fixed label was added to this pull request; all PRs with a changelog label need to have release notes included as part of the PR. If you haven't added release notes already, please do.

Release notes are added by creating a uniquely named file in the `.release-notes` directory. We suggest you call the file `47.md` to match the number of this pull request.

The basic format of the release notes (using markdown) should be:

```
## Title

End user description of changes, why it's important,
problems it solves etc.

If a breaking change, make sure to include 1 or more
examples what code would look like prior to this change
and how to update it to work after this change.
```

Thanks.

**Comment by SeanTAllen** [comment]
Please update the first comment here to be the commit message to use when we merge. 

First line should be a short sentence on what was fixed. 

Body should have additional detail.

**Comment by jackgene** [comment]
> Please remove the unrelated gitignore change.

Oh yeah, accidentally included that.

**Review by mfelsche — APPROVED** [review]

**Comment by SeanTAllen** [comment]
This shouldn't be merged until release notes are added.

**Comment by jackgene** [comment]
> This shouldn't be merged until release notes are added.

Will address this shortly, but I do have a question, per @ponylang-main:
> We suggest you call the file 47.md to match the number of this pull request.

But looking at the files in currently in `.release-notes`, they all seem to be version numbers. Should I name the file `0.4.2.md` (which seems like the natural next number)? Or should I update `next-release.md` and it'll automatically handle the rest? Or should I use `47.md` per @ponylang-main's suggestion?

Thank you, and sorry about the delays on this one.

**Comment by rhagenson** [comment]
Follow what ponylang-main says and call it 47.md

47 is the PR number.

The others are release entries. When a release is done the non-release entries are rolled together into a new release entry.

**Comment by jackgene** [comment]
Just added the release notes. Please let me know if you have any concerns with it.

**Comment by rhagenson** [comment]
I just took a look and do not see any issue with the release notes you added. This fixed a bug where the minimum of 1 second was incorrectly set to a minimum of 1 millisecond. All of that is now in the release notes.

File: .release-notes/47.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+## Make minimum default heartbeat 1000ms
+
+Default heartbeat value calculation updated to reflect code comment:
+```
+// use a quarter of the actual configured timeout but at minimum 1 second
+```
+(was defaulting to 1ms before)
```
Comment (SeanTAllen) [inline .release-notes/47.md:7]:
```suggestion
## Updated default connection heartbeat length

Previously, the default connection heartbeat was incorrectly set to 1ms. We've updated it to the value it was intended to be: 1000ms.

Sending a heartbeat every millisecond was excessive, and this change should improve performance slightly.
```

**Comment by SeanTAllen** [comment]
I've updated the PR title to better work in the CHANGELOG.

I've updated the release notes to be less "commit message" and more "release notes". Feel free to ask any questions about the changes so you can adapt any future titles etc as you make additional contributions.

**Pull request merged**